src/pybamm/solvers/summary_variable.py

```diff
@@ -69,7 +72,7 @@ def _initialize_for_cycles(self, cycle_summary_variables: list[SummaryVariables]
         self.first_state = None
         self.last_state = None
         self.cycles = cycle_summary_variables
-        self.cycle_number = np.arange(1, len(self.cycles) + 1)
+        self.cycle_number = np.arange(1, len(self.cycles) + 1, dtype=float)
```

Comment:
Switching from int to float, or float32 to float64 comes with a memory cost.
```suggestion
        self.cycle_number = np.arange(1, len(self.cycles) + 1)
```